### ISSUE TITLE
[Issue]: Cannot delete users

### ISSUE THREAD

**Issue by sypion** [jellyfin/jellyfin#11854]
### Please describe your bug

When attempting to delete a user from my Jellyfin server it does not proceed and continuously shows the loading animation.

### Reproduction Steps

1. Visit the `Users` tab in your admin dashboard.
2. Find the user you'd like to delete from your Jellyfin server
3. Click on `Delete`

### Jellyfin Version

10.9.0

### if other:

10.9.3

### Environment

```markdown
- OS: Linux (Ubuntu) x64
- Linux Kernel: 5.15.0-107-generic
- Virtualization: KVM
- Clients: Browser, Windows Client
- Browser: Arc (Chromium)
- FFmpeg Version: 6
- Reverse Proxy: No
- Base URL: localhost
- Networking: Host
- Storage: local
```

### Jellyfin logs

```shell
[2024-05-27 11:20:54.192 -07:00] [ERR] Error processing request. URL "DELETE" "/Users/a8d22bd83f9840c7b3e263d6b3f51e8f".
System.NullReferenceException: Object reference not set to an instance of an object.
   at Emby.Server.Implementations.Playlists.PlaylistManager.GetUserPlaylists(Guid userId)
   at Emby.Server.Implementations.Playlists.PlaylistManager.RemovePlaylistsAsync(Guid userId)
   at Jellyfin.Api.Controllers.UserController.DeleteUser(Guid userId)
   at lambda_method2405(Closure, Object)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(ActionContext actionContext, IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Jellyfin.Api.Middleware.ServerStartupMessageMiddleware.Invoke(HttpContext httpContext, IServerApplicationHost serverApplicationHost, ILocalizationManager localizationManager)
   at Jellyfin.Api.Middleware.WebSocketHandlerMiddleware.Invoke(HttpContext httpContext, IWebSocketManager webSocketManager)
   at Jellyfin.Api.Middleware.IPBasedAccessValidationMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager)
   at Jellyfin.Api.Middleware.LanFilteringMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager, IServerConfigurationManager serverConfigurationManager)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Api.Middleware.QueryStringDecodingMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.ReDoc.ReDocMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Api.Middleware.RobotsRedirectionMiddleware.Invoke(HttpContext httpContext)
   at Jellyfin.Api.Middleware.LegacyEmbyRouteRewriteMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.InvokeCore(HttpContext context)
   at Jellyfin.Api.Middleware.ResponseTimeMiddleware.Invoke(HttpContext context, IServerConfigurationManager serverConfigurationManager)
   at Jellyfin.Api.Middleware.ExceptionMiddleware.Invoke(HttpContext context)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by TimGels** [comment]
are you on 10.9.0 or on 10.9.3?

**Comment by sypion** [comment]
> are you on 10.9.0 or on 10.9.3?

10.9.3

**Comment by Shadowghost** [comment]
I am unable to reproduce this - does this user have anything specific? Like playlists assigned or similar?

**Comment by TimGels** [comment]
I am unable to reproduce with a standard user as well.

**Comment by TimGels** [comment]
I created a test user. And with that user I created a public playlist and a private playlist. I then proceeded to delete the test user, which got deleted without errors.

**Comment by sypion** [comment]
There are no playlists on my server from what I can see. This issue is currently present for all users. Creating a standard test user and attempting to delete it still results in the same issue.

**Comment by TimGels** [comment]
Are you trying this on web? If not, can you try it in a web browser?

**Comment by sypion** [comment]
> Are you trying this on web? If not, can you try it in a web browser?

Tried on the web and the problem still occured.

However, good news! After running the `Clean up collections and playlists` task and restarting the Jellyfin server, I am now able to delete users. Not sure what happened there.

**Comment by TimGels** [comment]
I Assume certain events caused the data to be borked in the database. I wonder how that can be replicated.

**Comment by Shadowghost** [comment]
THis migth be reproducible by adding a playlist to the user, adding an item, removing the item from the library (not the playlist) and then deleting the user.

**Comment by BotBlake** [comment]
> THis migth be reproducible by adding a playlist to the user, adding an item, removing the item from the library (not the playlist) and then deleting the user.

I tried reproducing this issue by breaking all sorts of things in Playlists and then deleting the User.
I've been sitting here any trying to break Playlists for an hour now.

Tried everything from weird names over deleting the media over deleting the whole library over messing with the Playlist.xml files.

Yet no matter what I did - The User that owned the Playlist was happyly removed through the Admin dashboard.

**Comment by sypion** [comment]
An odd thing I noticed before upgrading to 10.9 initially (although I am on 10.9.3 now) is that Jellyfin kept creating a playlist from an album I had that no one, including myself, was intentionally trying to make.

This [reddit post](https://www.reddit.com/r/jellyfin/comments/xtdhpt/playlists_i_didnt_make_keep_showing_up_with/) from 2 years ago is what helped me solve that problem. On Jellyfin web, I had deleted the playlist, then manually went in the album's directory on my host and deleted the misc. file that was causing Jellyfin to keep auto-creating a playlist.

Since then, I haven't tried deleting a user until today. I _have_ restarted Jellyfin multiple time since, so the `Clean up collections and playlists` task should have triggered and ran successfully. Not sure why this time it helped to run it manually again.

**Comment by TimGels** [comment]
It was a m3u file that you deleted?

**Comment by sypion** [comment]
If I recall, I believe it was an `.nfo` file.

**Comment by xd003** [comment]
I am also facing this issue rn on 10.9.4

**Comment by Brramble** [comment]
Im no 10.9.6 and also now seeing this issue.

`[2024-06-08 10:57:24.922 +01:00] [ERR] [31] Jellyfin.Api.Middleware.ExceptionMiddleware: Error processing request. URL "DELETE" "/Users/b2a1bcacb3e744718c79f092bcf048fa".
System.NullReferenceException: Object reference not set to an instance of an object.
   at Emby.Server.Implementations.Playlists.PlaylistManager.GetUserPlaylists(Guid userId)
   at Emby.Server.Implementations.Playlists.PlaylistManager.RemovePlaylistsAsync(Guid userId)
   at Jellyfin.Api.Controllers.UserController.DeleteUser(Guid userId)
   at lambda_method1490(Closure, Object)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(ActionContext actionContext, IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)`

**Comment by fuomag9** [comment]
Seeing this one as well on 10.9.6

```
jellyfin  | [13:44:31] [ERR] [129] Jellyfin.Api.Middleware.ExceptionMiddleware: Error processing request. URL DELETE /Users/censor.
jellyfin  | System.NullReferenceException: Object reference not set to an instance of an object.
jellyfin  |    at Emby.Server.Implementations.Playlists.PlaylistManager.GetUserPlaylists(Guid userId)
jellyfin  |    at Emby.Server.Implementations.Playlists.PlaylistManager.RemovePlaylistsAsync(Guid userId)
jellyfin  |    at Jellyfin.Api.Controllers.UserController.DeleteUser(Guid userId)
jellyfin  |    at lambda_method1500(Closure, Object)
jellyfin  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(ActionContext actionContext, IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
jellyfin  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
jellyfin  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
jellyfin  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
jellyfin  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
jellyfin  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
jellyfin  | --- End of stack trace from previous location ---
jellyfin  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
jellyfin  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
jellyfin  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
jellyfin  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
jellyfin  | --- End of stack trace from previous location ---
jellyfin  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
jellyfin  |    at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
jellyfin  |    at Jellyfin.Api.Middleware.ServerStartupMessageMiddleware.Invoke(HttpContext httpContext, IServerApplicationHost serverApplicationHost, ILocalizationManager localizationManager)
jellyfin  |    at Jellyfin.Api.Middleware.WebSocketHandlerMiddleware.Invoke(HttpContext httpContext, IWebSocketManager webSocketManager)
jellyfin  |    at Jellyfin.Api.Middleware.IPBasedAccessValidationMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager)
jellyfin  |    at Jellyfin.Api.Middleware.LanFilteringMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager, IServerConfigurationManager serverConfigurationManager)
jellyfin  |    at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
jellyfin  |    at Jellyfin.Api.Middleware.QueryStringDecodingMiddleware.Invoke(HttpContext httpContext)
jellyfin  |    at Swashbuckle.AspNetCore.ReDoc.ReDocMiddleware.Invoke(HttpContext httpContext)
jellyfin  |    at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
jellyfin  |    at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
jellyfin  |    at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
jellyfin  |    at Jellyfin.Api.Middleware.RobotsRedirectionMiddleware.Invoke(HttpContext httpContext)
jellyfin  |    at Jellyfin.Api.Middleware.LegacyEmbyRouteRewriteMiddleware.Invoke(HttpContext httpContext)
jellyfin  |    at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.InvokeCore(HttpContext context)
jellyfin  |    at Jellyfin.Api.Middleware.ResponseTimeMiddleware.Invoke(HttpContext context, IServerConfigurationManager serverConfigurationManager)
jellyfin  |    at Jellyfin.Api.Middleware.ExceptionMiddleware.Invoke(HttpContext context)
```

**Comment by fuomag9** [comment]
> > Are you trying this on web? If not, can you try it in a web browser?
> 
> Tried on the web and the problem still occurred.
> 
> However, good news! After running the `Clean up collections and playlists` task and restarting the Jellyfin server, I am now able to delete users. Not sure what happened there. Thank you both for your help!
> 
> Edit: That task is set to run every time on startup. Changed it to daily at a certain time so I don't have to worry about it.

This seems to have worked for me as well

**Comment by phito** [comment]
`PlaylistManager.cs` has a `#nullable disable`  declaration, which lead to these `NullReferenceException`  when dependencies are return null values (LibraryManager in this case). I'm looking into removing the `#nullable disable`  but this is my first time trying to contribute :)

**Comment by maddoglee** [comment]
I had a similar problem that had been bugging me for ages where I couldn't delete two users. 
This was my error in the logs.

`[2024-08-21 11:03:17.596 +10:00] [ERR] [31] Jellyfin.Api.Middleware.ExceptionMiddleware: Error processing request. URL "DELETE" "/Users/0b17ccc5786746ecafbcce225e772897".
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while saving the entity changes. See the inner exception for details.
 ---> Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 11: 'database disk image is malformed'.`

My database was corrupted. I had to use SQLITE to dump the db and then create a new one by reading it in. I am using Windows for my server. The best thing was to navigate to the location of the database and use commands there. I used ChatGPT to help me with the commands. After I repaired the db I was able to delete the users with no problem. 

I hope this helps someone!

This SQLITE command gave me errors in the db, then after i fixed it, it was 'ok'

`PRAGMA integrity_check;`
